Web/src/components/games/PandaSequence/useSeqGenerator.ts

```diff
@@ -0,0 +1,46 @@
+import { useRef, useCallback } from 'react';
+import seedrandom from 'seedrandom';
+import { Sequence } from './Sequence';
+
+const INITIAL_LENGTH = 2; // length of each memory sequence
+const INITIAL_TIMESTEP = 1000; // duration of each timestep
+const LENGTH_FACTOR = 1;
+const TIMESTEP_FACTOR = 0.8;
+
+function* randomWithinBounds(seed: string, lowerBound: number, upperBound: number)
+  : Generator<number, number, void> {
+  const random = seedrandom(seed);
+  while (true) yield Math.floor(random() * (upperBound - lowerBound) + lowerBound);
+}
+
+/**
+ * Takes in a seed and returns randomly generated Sequences of numbers between
+ * the range of lowerBound and upperBound.
+ */
+export default function useSeqGenerator(
+  seed = Date.now().toString(),
+  lowerBound = 1,
+  upperBound = 5,
+) {
+  const generator = useRef(randomWithinBounds(seed, lowerBound, upperBound)).current;
+  const sequence = useRef<Sequence | null>(null);
```

Comment:
Not sure why use `useRef` instead of `useState`

Reply:
Ah the reason I had to use useRef was because in the generate function on line 28, I need a reference to sequence rather than the sequence initialised on first render. It would perpetually be null if useState is used

Reply:
👍